### PR TITLE
fix(release-notes): remove element symbols and names and update design

### PR DESCRIPTION
**Pull request by brandyscarney**
This pull request updates the release notes at https://ionicframework.com/docs/reference/release-notes to do the following:
- Deletes the unused `scripts/bak/build-data/release-notes.ts` file (you can see that the `scripts/release-notes.js` file is what is used by the [npm scripts](https://github.com/ionic-team/ionic-docs/blob/55b72122873cd91260448c6cecfad835e13e0f97/package.json#L14)).
- Removes the element names that get added to the major/minor release headings
- Removes the element symbol on the left of releases
- Removes the double heading that shows up due to the heading in the release body
- Adds console warns for when the release notes have no releases
- Adds a console error when the script fails to get the releases
- Replaces the [`url.format`](https://github.com/ionic-team/ionic-docs/pull/2921/files#diff-e5b6dd7a973ee9fc0a90764c79e4d0e4da59c3616065385222145cd919ee47dfR27) code as it was [deprecated](https://github.com/orgs/nodejs/discussions/23694)
- Updates the design of the page & improves dark mode

<table>
  <thead>
    <td>
      <a href="https://ionicframework.com/docs/reference/release-notes">Before</a>
    </td>
    <td>
      <a href="https://ionic-docs-git-fw-3870-ionic1.vercel.app/docs/reference/release-notes">After</a>
    </td>
  </thead>
  <tr>
    <td valign="top">
      
    </td>
    <td valign="top">
      
    </td>
  </tr>
</table>

Design changes were approved by Ben

### CODE REVIEW

**Comment by brandyscarney** [comment]
@sean-perkins I actually thought about this but when I ran axe it flagged the menu and a ton of other areas for contrast issues so I figured it would be something we brought up later when we move to the common themes. This design was originally created by Ben so I didn't want to change it.

**Review by jaredcbaum — APPROVED** [review]

File: src/components/page/reference/ReleaseNotes/index.tsx [hunk 1]
```diff
@@ -6,17 +6,21 @@ import styles from './styles.module.css';
...
   version: string;
 }
 
 export default function ReleaseNotes(props: { [key: string]: any }) {
   if (releases.length === 0) {
+    console.warn('Could not load release notes data. Make sure that you have a valid GITHUB_TOKEN. ');
+    console.warn('Create a personal access token by following the below guide:');
+    console.warn('https://docs.github.com/en/enterprise-cloud@latest/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token');
+    console.warn('and then authorize it to work with SSO:');
+    console.warn('https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/authorizing-a-personal-access-token-for-use-with-saml-single-sign-on');
```
Comment (amandaejohnston) [inline src/components/page/reference/ReleaseNotes/index.tsx:22]: Nitpick: Could we combine these into one warning? If you use \`template strings\`, you can include line breaks.

**Review by amandaejohnston — APPROVED** [review]

File: src/components/page/reference/ReleaseNotes/index.tsx (hunk 1 above)
Comment (brandyscarney) [reply src/components/page/reference/ReleaseNotes/index.tsx]: Yeah, absolutely! Done here: https://github.com/ionic-team/ionic-docs/pull/2921/commits/5f195f6b3951ae9044f563d09e78bca2f3240e78

**Comment by brandyscarney** [comment]
FW-4116 has been created to address the axe contrast violations at a later time.

**Pull request merged**